### ISSUE TITLE
Improve default workflows naming

### ISSUE THREAD

**Issue by pcm32** [phnmnl/container-galaxy-k8s-runtime#51]
Currently we are providing a number of workflows as part of the base installation. We need to improve name and annotation so that they are more descriptive. The current case looks like this:

| Name                                                                                             | Annotation |
|--------------------------------------------------------------|------------|
| Reconstructed W4M sacurine workflow (imported from API)   |            |
| Fluxomics stationary with viz (imported from API)                    |            |
| xcmsCameraWorkflow (imported from uploaded file)               |            |
| MetFragWorkflow (imported from API)                                       |            |
| NMR MTBLS1 (imported from API)                                              |            |
| fluxomics-stationary-0.2 (imported from API)                            |            |
| LC-MS Preprocessing with W4M Tools (imported from API)     |            |

any suggestions of what should go as a more descriptive name and annotation for each workflow. Thanks!

**Comment by rsalek** [comment]
How about replacing "NMR MTBLS1 (imported from API)" with "1D NMR data processing workflow (data imported from MetaboLights using API)"

**Comment by pcm32** [comment]
Like that, thanks @rsalek! will add it to the table.

**Comment by pcm32** [comment]
Current result:

| Name                                                                                             | Annotation |
|--------------------------------------------------------------|------------|
| Reconstructed W4M sacurine workflow (imported from API)   |            |
| Fluxomics stationary with viz (imported from API)                    |            |
| xcmsCameraWorkflow (imported from uploaded file)               |            |
| MetFragWorkflow (imported from API)                                       |            |
| **1D NMR data processing workflow for MetaboLights studies**   |            |
| fluxomics-stationary-0.2 (imported from API)                            |            |
| LC-MS Preprocessing with W4M Tools (imported from API)     |            |

Please modify here.

**Comment by michaelvanvliet** [comment]
xcmsCameraWorkflow to: LC-MS Preprocessing: compound spectra extraction and annotation using XCMS and CAMERA

**Comment by ethevenot** [comment]
Reconstructed W4M sacurine workflow (imported from API) -> univariate-multivariate-statistics-w4m

**Comment by DSchober** [comment]
Hi guys,
 in case you like to follow a more formal approach and create a Workflow naming convention, here some initial thoughts. I thought Rachel Spicer did some work on this topic too?

There are no WF naming conventions established yet. there are merely some specialized task aligned recommendations i.e. https://kb.informatica.com/faq/7/Pages/17/503199.aspx
I have drafted some naming patterns compositions (list of entities likely appeariung for our use case below) and might submit these to http://ontologydesignpatterns.org/wiki/Category:NamingOP when sufficiently stable.
To come uop with a more robust pattern, I need to know the constraints for the WF name field in Galaxy, e.g. is there a char limitation, i.e. in terms of Stringsize or allowed characters?
After all, the pattern constitution totally depends on the way we can and intent to search for WFs. Also here, more guide on the search & retrieval and envisioned search interface would be nice.
For the time being a general into to formal naming conventions is at
https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-10-125 and its an established OBO Foundry policy (http://www.obofoundry.org/principles/fp-012-naming-conventions.html)

Potential Naming Pattern constituents should probably be general enough to allow for abstract searching, but specific enough to be intuitive and flexible. To encourage re-usability thy should probably not contain the dataset descriptions, but again this depends on the envisioned usage.

Elements for a potential WF Naming Pattern:
WF ID or Source (W4M, PhenoMeNal, ...)
(DomainName (metabolomics, proteomics, transcriptomics, genomics))
(TechnologyName (mass spec, NMR, fluxomics, …))
AssayMethodName (LCMS, 1D NMR, …)
ProcessingToolNames (XCMS,Camera, Metfrag, Batman, … ) e.g. from BioTools.org
ProcessingCharacteristics, e.g. ProcessingStage (preprocessing, postprocessing, statistics, assignment, …) this will be most variant
Dataset/Probe/SampleName (MTBLS1, sacurine, …)e.g. coming from Database IDs

E.g. this would lead to the following WF dataset independent) names:
Etiennes WF: Reconstructed W4M sacurine workflow (imported from API) -> **W4M LCMS XCMS CAMERA univariate-multivariate-statistics** 
Mikes: LC-MS Preprocessing: compound spectra extraction and annotation using XCMS and CAMERA -> **LCMS XCMS CAMERA compound spectra extraction and annotation**
Rezas/ilverolucas:1D NMR data processing workflow for MetaboLights studies-> **MTBLS1 1D NMR metabolightsimport, nmrMLConv rnmr1d data processing and plot**
...
But again, at this moment it is difficult to make an informed decision on an actual WF naming pattern.
Let me know if you are interested in such a more formal naming convention.

**Comment by c-ruttkies** [comment]
MetFragWorkflow -> LC-MS/MS Metabolite Annotation: MS/MS compound spectra extraction, annotation and identification using XCMS, CAMERA and MetFrag

**Comment by c-ruttkies** [comment]
or with the naming convention of @DSchober it should be like:
PHNMNL XCMS CAMERA MetFrag MS/MS compound spectra extraction and annotation

**Comment by pcm32** [comment]
I like @DSchober proposal very much! Besides length restrictions, anything else you need to know? Does it help the fact that there is an additional field called 'Annotation' that we can use?

**Comment by pcm32** [comment]
As a reference, situation seems to be pretty nosiy at the main Galaxy site:

https://usegalaxy.org/workflow/list_published

**Comment by DSchober** [comment]
Aaargh...  ! But the most uninformative names (in that order) are:

Workflow constructed from history 'Unnamed history' _(aah THAT workflow...)_
unnamed workflow
new work _( ...and already old)_
workflow.vcf _(at least ext hints for genomics)_
Final workflow _(I like the fatalistic attitude, mate!)_
Galaxy workflow _(slightly underspecified)_
from the beginning
full
Yale_May2012 _(someone very proud of his/her affiliation?)_
Coursera project
TopHat
op  _( ...of cause)_

... they deserve a price.

The first one is autogenerated (somewhat excused) though. Many are toy-examples such as test, demo, myworkflow, and 101 expls. They should really clean up this mess and either allow a playground/sandbox, or filter the visible WFs by quality criteria. Or not encourage people to share their dummy WFs in the first place.

What is the size limit for this WF name field ? Does it allow spaces ? Does it prevent certain Characters? I see frequent usage of a colon. Also Mike used it above. How is it used ? Do you separate a general WF abstraction from the more specific tool related process steps that way ?
 Good to see that the WFs that were more thoroughly annotated by Community tags are also the ones with more clear names. Not sure how to access this 'Community Tag' list? Or is it feetext as well? The additional Annotation field surely helps.

I will try to work something out in a more formal way i.e. indicating cardinalities of pattern parts as well once I have played around with the search interfaces.

**Comment by pcm32** [comment]
@DSchober do you think that you could come up with one or two naming patters from where partners can choose from before the 10th of August please? So that we can have names in place before the release on the 16th and we can update the documentation. Thanks!

**Comment by pcm32** [comment]
Ok, according to @DSchober we use then this pattern in the mean time:

- WF ID or Source (W4M, PhenoMeNal, ...)
- (DomainName (metabolomics, proteomics, transcriptomics, genomics))
- (TechnologyName (mass spec, NMR, fluxomics, …))
- Assay Method Name (LCMS, 1D NMR, …)
- Processing Tool Names (XCMS,Camera, Metfrag, Batman, … ) e.g. from BioTools.org
- Processing Characteristics, e.g. ProcessingStage (preprocessing, postprocessing, statistics, assignment, …)

This means:

**Comment by pcm32** [comment]
| Name                                                                                             | Annotation |
|--------------------------------------------------------------|------------|
| Reconstructed W4M sacurine workflow (imported from API)   |        W4M univariate-multivariate-statistics    |   
| Fluxomics stationary with viz (imported from API)                    |      PHNMNL 13C-MS-Fluxomics ramid-midcor-iso2flux with viz      |
| MetFragWorkflow (imported from API)                                       |      PHNMNL LCMS/MS XCMS CAMERA MetFrag compound spectra extraction and annotation      |
| 1D NMR data processing workflow for MetaboLights studies   |       PHNMNL 1D NMR nmrMLConv rnmr1d data processing and plot     |
| fluxomics-stationary-0.2 (imported from API)                            |     PHNMNL 13C-MS-Fluxomics ramid-midcor-iso2flux       |
| LC-MS Preprocessing with W4M Tools (imported from API)     |       W4M LCMS XCMS CAMERA univariate-multivariate-statistics     |

**Comment by DSchober** [comment]
Hi, 
I thought the naming pattern would actually be used for the NAME field, rather than the ANNOTATION field? 
We could state that the Annotation field could constitute an own short name and a longer freetext describung the WF in own words...

For the name pattern, I think for PhenomeNal the 'PHNMNL' prefix (WF source) can be omitted as is implied here ( kinda default in our realm here). Just indicate this when not coming from PHNMNL, e.g. for W4M case.

Hence, I would suggest the following:
NAME:
(MTBLS404) W4M metabolomics MS or NMR ISATab R biosigner univariate-multivariate-statistics
Fluxomics 13C-MS ramid midcor iso2flux [here add what is done by the software] with viz
metabolomics (or small molecule analytics) LCMS/MS XCMS CAMERA MetFrag compound spectra extraction and annotation
(MTBLS) metabolomics 1D NMR nmrMLConv rnmr1d data processing and plot
Fluxomics 13C-MS  ramid midcor iso2flux  [here add what is done by the software] 
W4M metabolomics LCMS XCMS CAMERA univariate-multivariate-statistics
(this last WF I am not sure about, could you send me a link to documentation ?

In general I would either get rid of the hyphens or use them consistently i.e. either to separate the tool names(prenultimate bit), OR Functions carried out in a WF (endbit).

**Comment by DSchober** [comment]
In the Annotation field users should be able to paste a longer free text description verbally describing their WF, i.e. as appearing in a paper methods section. This is also the place to add stuff of marginal interest that is less likely suitable as search attribute for WFs.
(I can today only react on email and will check that at 7 pm CET.)

**Comment by DSchober** [comment]
As a domain name more general than metabolomics I suggest 'small molecule analytics' as this encompasses metabolomics and environmental analytics.

**Comment by pcm32** [comment]
Apologies @DSchober my use of column headers was incorrect... please see table now.

**Comment by pcm32** [comment]
Ok, so newer names to the best that I could understand @DSchober's names above, better now?:

| Old name                                                                                             | Newer names |
|--------------------------------------------------------------|------------|
| Reconstructed W4M sacurine workflow (imported from API)   |        W4M metabolomics MS or NMR ISATab R biosigner univariate/multivariate statistics   |   
| Fluxomics stationary with viz (imported from API)                    |      Fluxomics 13C-MS ramid midcor iso2flux with viz      |
| MetFragWorkflow (imported from API)                                       |      metabolomics LCMS/MS XCMS CAMERA MetFrag compound spectra extraction and annotation      |
| 1D NMR data processing workflow for MetaboLights studies   |       metabolomics 1D NMR nmrMLConv rnmr1d data processing and plot     |
| fluxomics-stationary-0.2 (imported from API)                            |     Fluxomics 13C-MS ramid midcor iso2flux       |
| LC-MS Preprocessing with W4M Tools (imported from API)     |       W4M metabolomics LCMS XCMS CAMERA univariate/multivariate statistics     |

**Comment by rsalek** [comment]
We need a short name too, some of them are too long to pronounce!

**Comment by rsalek** [comment]
Compromise would be  "Metabolomics 1D NMR MetaboLights data processing and plot"

**Comment by ethevenot** [comment]
Dear all,
I fully agree that a rigorous naming of the workflows is required. However, I think that the name should be also meaningful for the user. Regarding the statistical workflow, I think that the current name (W4M Metabolomics MS or NMR ISATab R biosigner univariate/multivariate statistics) should be modified since:
1) the workflow is not restricted to metabolomics data (any omics)
2) it is not restricted to MS nor NMR (any technology)
3) I do not understand what ISATab nor R means here
4) the included 'biosigner' module name is not very meaningful here ('feature selection' would be more appropriate if needed).

So I would suggest something more like this:
W4M omics univariate/multivariate statistics and feature selection

What do you think?

Etienne.

**Comment by pcm32** [comment]
I agree that having all the modules inside the workflow name is too long, I would go for at the most one or two relevant tools, if at all mentioning tools. We could add tools as tokens to the annotation field maybe (as tags).

**Comment by DSchober** [comment]
OK, then only cite main Tool, but cite all Tools in the Annotation. We could add a convention for the Annotation, i.e. to enforce the two parts User Preferred (short/intuitive) WF name + more extensive freetextual definition.  Etienne, me and Christoph R looked at the WF and thought it uses the Tools ISATab R and Biosigner. I am not sure as I have not seen docs. 
I would recommend to state the main domain name that this WF was created for, as We would end up with many many statistics WFs that are of cause generally applicable, but people are lazy and like task aligned WFs with a plethora of fitting parameters I guess.

**Comment by DSchober** [comment]
Further elements appearing in the Annotation could be used SourceDatabaseIDs (MTBLS404), data types/standards (ISATab), SampleSource (Human Urine?)...
This all depends on what a user is expected to look for as search attributes when trying to find a WF. For that I think the combination/succession of tools/modules applied is really essential (not just one, as this will make many names arbitrary I recon).

**Comment by DSchober** [comment]
@rsalek , in your
"Compromise would be "Metabolomics 1D NMR MetaboLights data processing and plot"
or
WF ID or Source (W4M, PhenoMeNal, ...)
(DomainName (metabolomics, proteomics, transcriptomics, genomics))
(TechnologyName (mass spec, NMR, fluxomics, …))
Assay Method Name (LCMS, 1D NMR, …)
Main Processing Tool Name"
... suggestion, I feel your name is inconsistent with your pattern, as it laks the tool names. 
I still feel we should stick to a formal pattern in the name.

**Comment by DSchober** [comment]
@ethevenot , can you live with a pattern compliant:
"W4M Metabolomics MS or NMR R biosigner univariate/multivariate statistics" ?
I assume the statistics is done weith R and biosigner tools. IsaTab was the format and might be put in the Annotation field. 
@salek, can you live with a pattern compliant:
"Metabolomics 1D NMR nmrMLConv rnmr1d metabolights data processing and plot"
You can have the short intuitive name in the Annotation, but I feel people should see the tools used in a WF, even if th eformal names get a little longer. I feel they are still manageable.

**Comment by DSchober** [comment]
Another idea would be to extract a semi-formal name from the Galaxy module names, i.e. for Rezas WF: mtbls_nrmr_raw_importer nmrMLconv Zip_nmrML_Collection_rnmr1d rnmr1d_stacked_plot.
Keep in mind that we would have a separator convention (for module names use the underscore, (this would allow us to use the space in the WF name to separate modules).

**Comment by djcomlab** [comment]
For an informal/short name, surely we can just go with something that's not so structured and just more like an arbitrary label (that us as humans are happy with). e.g.

W4M Omics Feature Selection
Fluxomics Workflow 1
MetFrag Workflow
1D NMR Processing for MetaboLIghts
Fluxomics Workflow 2
W4M LC-MS Feature Selection
etc.

The devil in the detail would be in the long name or annotations of course, but a short name in my opinion should just be as usable as possible (and potential name clashes dealt with when new people try and add their workflows - they will just have to pick something else if one is taken, tough bananas).

So basically: short name - no naming pattern or convention. For the rest - pack in as much as needed into the annotation/long name etc.

**Comment by DSchober** [comment]
I have shifted the 'fluxomics' indicator to the DomainName Element as its not really a certain technology or assay. I have added the new pattern proposal under https://docs.google.com/spreadsheets/d/172P_sVkZhhDWOe_t4BOtNdm86P5iFJNH3I4It2H50DQ/edit#gid=1458978189
As now the Technology Name is redundant with Assay Method Name, I have removed it, so the new pattern is shorter.

**Comment by rsalek** [comment]
I'm looking at other and seems many follow and informal routes, which does not seem nice overtime. 
e.g.
https://usegalaxy.org/workflow/list_published
I would think we can compromise here:

> cite main Tool

> the workflow is not restricted to metabolomics data (any omics)
it is not restricted to MS nor NMR (any technology)
I do not understand what ISATab nor R means here
the included 'biosigner' module name is not very meaningful here ('feature selection' would be more appropriate if needed).

"W4M Omics MS or NMR biosigner univariate/multivariate statistics and feature selection"
to (assay=generic)
"W4M Omics generic biosigner feature selection statistics"

"Metabolomics NMR_rnmr1d MetaboLights data processing and plot" removed nmrMLConvert as it could be optional 
"Fluxomics 13C-MS_iso2flux with visualization"
"Fluxomics 13C-MS_iso2flux without visualization"
"Metabolomics LCMS/MS_XCMS CAMERA MetFrag processing and annotation"

**Comment by pcm32** [comment]
Since these have been incorporated into the release, I will close this. We can certainly re-float the discussion later on.
